tests/e2e/module_unmanage_test.go

```diff
@@ -0,0 +1,94 @@
+package e2e_test
+
+import (
+	templatev1alpha1 "github.com/kyma-project/template-operator/api/v1alpha1"
+
+	"github.com/kyma-project/lifecycle-manager/api/shared"
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	. "github.com/kyma-project/lifecycle-manager/pkg/testutils"
+)
+
+var _ = Describe("Unmanaging Kyma Module", Ordered, func() {
+	kyma := NewKymaWithSyncLabel("kyma-sample", ControlPlaneNamespace, v1beta2.DefaultChannel)
+	InitEmptyKymaBeforeAll(kyma)
+	CleanupKymaAfterAll(kyma)
+
+	module := NewTemplateOperator(v1beta2.DefaultChannel)
+
+	Context("Given SKR Cluster", func() {
+		It("When Kyma Module is enabled on SKR Kyma CR", func() {
+			Eventually(EnableModule).
+				WithContext(ctx).
+				WithArguments(skrClient, defaultRemoteKymaName, RemoteNamespace, module).
+				Should(Succeed())
+		})
+
+		It("Then Module Operator is deployed on SKR cluster", func() {
+			Eventually(CheckIfExists).
+				WithContext(ctx).
+				WithArguments(ModuleResourceName, TestModuleResourceNamespace, "apps", "v1",
+					"Deployment", skrClient).
+				Should(Succeed())
+			By("And KCP Kyma CR is in \"Ready\" State")
+			Eventually(KymaIsInState).
+				WithContext(ctx).
+				WithArguments(kyma.GetName(), kyma.GetNamespace(), kcpClient, shared.StateReady).
+				Should(Succeed())
+		})
+
+		It("When Kyma Module is unmanaged", func() {
+			Eventually(SetModuleUnmanaged).
+				WithContext(ctx).
+				WithArguments(skrClient, defaultRemoteKymaName, RemoteNamespace, module.Name).
+				Should(Succeed())
+
+			By("And Module CR on SKR Cluster is not removed")
+			Consistently(CheckIfExists).
+				WithContext(ctx).
+				WithArguments(TestModuleCRName, RemoteNamespace, templatev1alpha1.GroupVersion.Group,
+					templatev1alpha1.GroupVersion.Version, string(templatev1alpha1.SampleKind), skrClient).
+				Should(Succeed())
+
+			By("And Module Operator Deployment is not removed on SKR cluster")
+			Consistently(CheckIfExists).
+				WithContext(ctx).
+				WithArguments(ModuleResourceName, TestModuleResourceNamespace,
+					"apps", "v1", "Deployment", skrClient).
+				Should(Succeed())
+
+			By("And Manifest CR is removed")
```

Comment:
I would put this check after `SetModuleUnmanaged`, then check module CR is not removed and so on.